kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/unmarshall/nodes/NodeConnector.java

```diff
@@ -517,15 +531,42 @@ private boolean isCentre(final double magnetRelativeX,
                 Math.abs((viewHeight / 2) - magnetRelativeY) < CENTRE_TOLERANCE;
     }
 
-    Node getNode(final JSIDMNEdge jsidmnEdge,
-                 final List<NodeEntry> entries) {
+    Node getSourceNode(final JSIDMNEdge jsidmnEdge,
+                       final List<NodeEntry> entries) {
 
         if (entries.size() == 1) {
             return entries.get(0).getNode();
         }
 
         final JSIPoint jsiSource = Js.uncheckedCast(jsidmnEdge.getWaypoint().get(0));
-        final Point2D source = new Point2D(jsiSource.getX(), jsiSource.getY());
+        final Point2D source = createPoint(jsiSource);
+
+        return getNodeFromPoint(source, entries);
+    }
+
+    Point2D createPoint(final JSIPoint point) {
+        return new Point2D(point.getX(), point.getY());
+    }
+
+    Node getTargetNode(final JSIDMNEdge jsidmnEdge,
+                       final List<NodeEntry> entries) {
+
+        if (entries.size() == 1) {
+            return entries.get(0).getNode();
+        }
+
+        final JSIPoint jsiTarget = Js.uncheckedCast(jsidmnEdge.getWaypoint().get(1));
```

Comment:
I know there is a jira asking an enhancement to add "bend points" for connections. Unfortunately I struggle to find the jira link.

I guess we use index `1` to access target waypoint, what will maybe get out of date with implementing such a feature. Maybe could be replaced by something like `.size() - 1`?

Just mentioning to consider. Not blocking merge.

Reply:
@jomarko here's the JIRA https://issues.redhat.com/browse/KOGITO-5119 - I agree `.size() - 1` would be more flexible, but I personally would keep it as is for now, as we will need to add other changes to support bend-points anyhow :-)